core/kazoo_number_manager/src/carriers/knm_carriers.erl

```diff
@@ -307,6 +323,43 @@ disconnect(Number) ->
             Number
     end.
 
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Create a number in a discovery state.
+%% @end
+%%--------------------------------------------------------------------
+-spec create_discovery(ne_binary(), module(), ne_binary(), kz_json:object()) ->
+                              knm_number_return().
+create_discovery(DID=?NE_BINARY, Carrier, ?MATCH_ACCOUNT_RAW(AuthBy), Data=?JSON_WRAPPER(_))
+  when is_atom(Carrier) ->
+    case knm_number:get(DID) of
+        {'ok', _Number}=Ok -> Ok;
```

Comment:
Yes. This is important. We may have a number in db that is actually available to buy (client released it without going through us). We are never the most up to date: the actual carrier is.

So, if we already have the number, we return it here. Somewhere else in knm_carriers the number will be updated (see transition_existing_to_discovery).